### ISSUE TITLE
Keras on Caffe2

### ISSUE THREAD

**Issue by abhinavvishnu** [facebookarchive/caffe2#852]
Hi Guys:

Is there a plan to support Keras on Caffe?

thanks,

:- Abhinav

**Comment by hgaiser** [comment]
Shouldn't the question be the other way around? (And thus asked to Keras devs? ;) )

**Comment by colincsl** [comment]
Hans, as someone who has used Keras extensively in the past and just started using Caffe2, just no ;). From a modeling standpoint Keras has a very sleek API and for the most part it has nice docs. 

I don't know if anyone focused on Brew is thinking about this, but adopting the Keras spec could get more people on the Caffe 2 bandwagon.

**Comment by hgaiser** [comment]
Thanks for the feedback, wasn't aware that Keras implementation worked both ways. I assumed (wrongly apparently) that Keras was only an abstraction on top of other APIs, didn't know it defined its own API to be implemented.

**Comment by slayton58** [comment]
Indeed, it looks like it's as simple (for some definition of simple) as defining a new backend ([TF backend here](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py))

**Comment by ThatAIGeek** [comment]
Hi there, so any plans to implement this?
